openshift-kube-apiserver/admission/customresourcevalidation/authentication/validate_authentication.go

```diff
@@ -132,3 +138,59 @@ func validateAuthenticationSpec(spec configv1.AuthenticationSpec) field.ErrorLis
 func validateAuthenticationStatus(status configv1.AuthenticationStatus) field.ErrorList {
 	return crvalidation.ValidateConfigMapReference(field.NewPath("status", "integratedOAuthMetadata"), status.IntegratedOAuthMetadata, false)
 }
+
+func validateOIDCProvider(path *field.Path, provider configv1.OIDCProvider) field.ErrorList {
+	errs := field.ErrorList{}
+	errs = append(errs, validateClaimMappings(path, provider.ClaimMappings)...)
+	return errs
+}
+
+func validateClaimMappings(path *field.Path, claimMappings configv1.TokenClaimMappings) field.ErrorList {
+	path = path.Child("claimMappings")
+	errs := field.ErrorList{}
+	errs = append(errs, validateUIDClaimMapping(path, claimMappings.UID)...)
+	errs = append(errs, validateExtraClaimMapping(path, claimMappings.Extra...)...)
+	return errs
+}
+
+func validateUIDClaimMapping(path *field.Path, uid *configv1.TokenClaimOrExpressionMapping) field.ErrorList {
+	if uid == nil {
+		return nil
+	}
+
+	if uid.Expression != "" {
+		err := validateCELExpression(&authenticationcel.ClaimMappingExpression{
+			Expression: uid.Expression,
+		})
+		if err != nil {
+			return field.ErrorList{field.Invalid(path.Child("uid", "expression"), uid.Expression, err.Error())}
+		}
+	}
+
+	return nil
+}
+
+func validateExtraClaimMapping(path *field.Path, extras ...configv1.ExtraMapping) field.ErrorList {
+	errs := field.ErrorList{}
+	for i, extra := range extras {
+		errs = append(errs, validateExtra(path.Child("extra").Index(i), extra)...)
+	}
+	return errs
+}
+
+func validateExtra(path *field.Path, extra configv1.ExtraMapping) field.ErrorList {
+	err := validateCELExpression(&authenticationcel.ExtraMappingExpression{
+		Key:        extra.Key,
+		Expression: extra.ValueExpression,
+	})
+	if err != nil {
+		return field.ErrorList{field.Invalid(path.Child("valueExpression"), extra.ValueExpression, err.Error())}
+	}
+
+	return nil
+}
+
+func validateCELExpression(accessor authenticationcel.ExpressionAccessor) error {
+	_, err := authenticationcel.NewDefaultCompiler().CompileClaimsExpression(accessor)
```

Comment:
Would it make sense to create only one constructor and reuse it to validate all CELs?

Reply:
Done in the latest commit